lang/csharp/src/apache/main/Reflect/DotnetProperty.cs

```diff
@@ -67,7 +67,8 @@ private bool IsPropertyCompatible(Avro.Schema.Type schemaTag)
                 case Avro.Schema.Type.Array:
                     return typeof(IEnumerable).IsAssignableFrom(propType);
                 case Avro.Schema.Type.Map:
-                    return typeof(IDictionary).IsAssignableFrom(propType);
+                    var dictionaryType = typeof(IDictionary);
+                    return dictionaryType.IsAssignableFrom(propType) && propType.GenericTypeArguments[0] == typeof(string);
```

Comment:
Good catch,  I thought it was checked prior to this but a quick unit test using Hashtable threw an Index Out of Range Exception.  That raises the question if this should actually be a check with typeof(IDictionary<,>) instead.

Reply:
Ideally if it implements IDictionary<TKey, TValue> and TKey can be reasonably converted to string then it would be supported.  Unfortunately its not very straight forward to determine what a type implements without checking all the interfaces on the type.   

I noticed ConfigurationBinder has a [FindOpenGenericInterface](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs#L732) method that seems to do the trick